Azure/azure-sdk-for-java
# Read many fix

# Description

This PR fixes the issue when a point-read based `404 / 0` response drops existing point-read based documents from the resultant `FeedResponse` instance in spite of the ID of these documents being specified in the `readMany` request. The expected behavior is to ignore non-existent documents but fetch other documents.

# Testing

Test: `CosmosItemTest#readManyWithMultiplePartitionsAndSome404s`

- Create a multi-partition container - preferably with more than 2 physical partitions.
- Ensure documents are created on each physical partition.
- Query 1 document per physical partition using the `FeedRange` instance for the physical partition. This is done to ensure when `readMany` is used to query the documents back, it only uses point-reads to do so.
- Choose _n_ documents in which a non-existent ID is injected.
- Read these documents back using the `readMany` API.
- Verify if the count of fetched documents through `readMany` equals (total no. of physical partitions) - _n_  
- Repeat the above steps by varying _n_ from 0 to the total no. of physical partitions.

# Benchmarking results

- The fix shows a marginal tail latency increase (in the order of 1 ms to 5 ms)
<img width="1016" alt="image" src="https://github.com/Azure/azure-sdk-for-java/assets/16476407/f157a720-435d-448a-acdc-72bf02b4f4b5">
